src/bots/telegram/client.js

```diff
@@ -47,16 +47,24 @@ const { BOT_TOKEN } = config;
 const client = new Telegraf(BOT_TOKEN);
 
 class Name {
-  static from(user: Telegram$User): string {
-    if (user.username) {
-      return user.username;
+  static from(author: Telegram$User | Telegram$Chat): string {
+    if (author.username) {
+      return author.username;
     }
 
-    if (user.last_name) {
-      return `${user.first_name} ${user.last_name}`;
+    if (author.title) {
+      return author.title;
     }
 
-    return user.first_name;
+    if (author.first_name) {
+      if (author.last_name) {
+        return `${author.first_name} ${author.last_name}`;
+      }
+
+      return author.first_name;
+    }
```

Comment:
I don't quite like the logic here. Let's change it so user with `last_name` but without `first_name` will also qualify. Something like

```js
const names = [author.first_name, author.last_name].filter(x => x);
if (names.length) {
  return names.join(' ');
}
```

Reply:
@ForNeVeR done 35a3068278e5a939ea07c7d652969374d13c9d5e